config/shared_config.go

```diff
@@ -272,6 +280,15 @@ type SharedConfig struct {
 
 	// aws sdk app ID that can be added to user agent header string
 	AppID string
+
+	IgnoreConfiguredEndpoints *bool
+
+	BaseEndpoint string
+
+	// Services section values
+	// {"serviceId": {"key": "value"}}
+	// e.g. {"s3": {"endpoint_url": "example.com"}}
+	Services map[string]map[string]string
```

Comment:
* The values in `[service whatever]` could be non-objects though (they're just regular sections with another name) so I think we have to generalize to `Services map[string]Section`
* [nonblocking] probably want to leave unexported for now.

Reply:
We actually _must_ not export this, `Section` is internal.

Reply:
re-reading the spec, its not actually clear whether the `services` section is a normal section. it seems to say that the properties in a `services` section are all service subproperties? if not though, its not a big change, just need to move the logic that knows how to parse it up, and should be a small lift.

Reply:
ok, instead of making the value a `Section`, i wrapped the map in a Services struct, and top-level non map values can be defined in this new struct. this follows a similar pattern with sso session sections. i dont think we should be propagating the `Section` to the layer above. all `Section` processing into `SharedConfig` fields seems like it should happen in `setFromIniSection`. will push this change to this branch soon

Reply:
Agreed, new way LGTM.